src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs

```diff
@@ -317,6 +317,20 @@ public static IResourceBuilder<T> WithLifetime<T>(this IResourceBuilder<T> build
         return builder.WithAnnotation(new ContainerLifetimeAnnotation { Lifetime = lifetime }, ResourceAnnotationMutationBehavior.Replace);
     }
 
+    /// <summary>
+    /// Sets the pull policy for the container resource.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">Builder for the container resource.</param>
+    /// <param name="pullPolicy">The pull policy behavior for the container resource.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithImagePullPolicy<T>(this IResourceBuilder<T> builder, ImagePullPolicy pullPolicy) where T : ContainerResource
```

Comment:
Are there tests we can write for this?

At a minimum at least calling this API and ensuring the annotation is there. Ideally, if we can tell what data gets sent to DCP, we could ensure the image pull policy is set correctly.

Reply:
Unit tests added